src/stan/model/indexing/lvalue.hpp

```diff
@@ -55,17 +56,17 @@ inline void assign(T& x, const nil_index_list& /* idxs */, U&& y,
  * Assign the specified standard vector rvalue to the specified
  * standard vector lvalue.
  *
- * @tparam T lvalue container element type
- * @tparam U rvalue container element type, which must be assignable to `T`
+ * @tparam Vec1 vector type to be assigned to
+ * @tparam Vec2 vector type that must be assignable to `Vec1`
```

Comment:
```suggestion
 * @tparam Vec2 vector type with sccalar that must be assignable to scalar of `Vec1`
```

Reply:
Alternatively you can revert all changes to this overload. With requires of the other one this one should be selected for vectors even in previous version.

Reply:
I think with your other suggestion this one is fine to leave as is